src/test/java/org/springframework/data/couchbase/repository/SimpleReactiveCouchbaseRepositoryDeleteAllIntegrationTests.java

```diff
@@ -0,0 +1,94 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.data.couchbase.repository;
+
+import com.couchbase.client.java.Bucket;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TestName;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.couchbase.ContainerResourceRunner;
+import org.springframework.data.couchbase.ReactiveIntegrationTestApplicationConfig;
+import org.springframework.data.couchbase.repository.config.ReactiveRepositoryOperationsMapping;
+import org.springframework.data.couchbase.repository.support.IndexManager;
+import org.springframework.data.couchbase.repository.support.ReactiveCouchbaseRepositoryFactory;
+import org.springframework.data.repository.core.support.ReactiveRepositoryFactorySupport;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.TestExecutionListeners;
+import reactor.core.publisher.Mono;
+
+import java.util.NoSuchElementException;
+
+import static junit.framework.TestCase.fail;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.springframework.data.couchbase.CouchbaseTestHelper.getRepositoryWithRetry;
+
+/**
+ * @author David Kelly
+ */
+@RunWith(ContainerResourceRunner.class)
+@ContextConfiguration(classes = ReactiveIntegrationTestApplicationConfig.class)
+@TestExecutionListeners(SimpleReactiveCouchbaseRepositoryListener.class)
+public class SimpleReactiveCouchbaseRepositoryDeleteAllIntegrationTests {
+    // We do these tests here, rather in the simple crud tests, so we don't interact
+    // with those tests that make assumptions about what documents should be
+    // in the repo.
+    @Rule
+    public TestName testName = new TestName();
+
+    @Autowired
+    private Bucket client;
+
+    @Autowired
+    private ReactiveRepositoryOperationsMapping operationsMapping;
+
+    @Autowired
+    private IndexManager indexManager;
+
+    private ReactiveUserRepository repository;
+
+    private long getCount() {
+        return repository.count().block().longValue();
+    }
+
+    @Before
+    public void setup() throws Exception {
+        ReactiveRepositoryFactorySupport factory = new ReactiveCouchbaseRepositoryFactory(operationsMapping, indexManager);
+        repository = getRepositoryWithRetry(factory, ReactiveUserRepository.class);
+    }
+
+    @Test
+    public void simpleDeleteAll() {
+        String key = "my_unique_user_key";
+        ReactiveUser instance = new ReactiveUser(key, "foobar", 22);
+        repository.save(instance).block();
+
+        // we put a doc in, lets be sure the count reflects that there is
+        // at least one doc in there...
+        assertTrue(getCount() > 0L);
+
+        repository.deleteAll().block();
+
+        try {
+            getCount();
```

Comment:
views are eventually consistent, so this might race at the first view attempts.. I think it's good for now, but keep in mind should you see it failing occasionally this might be the reason.

Reply:
ohh - great point.  Thx -- maybe I need to rethink this test a bit, when I update it to mixup the count issue...